frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -267,41 +275,87 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
             actions.stopAnimation()
         },
         setSpeed: ({ speed }) => {
-            values.replayer?.setConfig({ speed })
+            values.player?.replayer?.setConfig({ speed })
         },
-        seek: async ({ time, forcePlay }, breakpoint) => {
+        seek: async ({ playerPosition, forcePlay }, breakpoint) => {
             // Real seeking is debounced so as not to overload rrweb.
             await breakpoint(100)
+            actions.stopAnimation()
 
-            // Time passed into seek function must be timestamp offset time.
-            const nextTime = getZeroOffsetTime(time ?? 0, values.meta)
+            actions.setCurrentPlayerPosition(playerPosition)
 
-            // Set current time to keep player updated. Replayer will catch up once time is buffered
-            actions.setCurrentTime(time ?? 0)
+            // Check if we're seeking to a new segment
+            let nextSegment = null
+            if (playerPosition) {
+                nextSegment = getSegmentFromPlayerPosition(playerPosition, values.sessionPlayerData.metadata.segments)
+                if (
+                    nextSegment &&
+                    (nextSegment.windowId !== values.currentSegment?.windowId ||
+                        nextSegment.startTimeEpochMs !== values.currentSegment?.startTimeEpochMs)
```

Comment:
Might be a tad redundant since I don't see how this can happen  when `nextSegment.windowId !== values.currentSegment?.windowId`

Reply:
Oooo good catch, I think this is a bug. Should be an `&&`, not an `||`

Reply:
Oh wait, no this should be an `||`. It only hits the startTime check if the windowIds are equal.